omod/src/main/webapp/resources/scripts/conditionlist/controllers/editcondition.controller.js

```diff
@@ -0,0 +1,129 @@
+var app = angular.module("conditionApp",
+    ['uicommons.widget.coded-or-free-text-answer', 'app.restfulServices', 'app.models', 'app.commonFunctionsFactory']);
+
+app.controller("ConditionController", ConditionController);
+ConditionController.$inject = ['$scope', 'RestfulService', 'ConditionModel', 'ConceptModel', 'CommonFunctions'];
+
+function ConditionController($scope, RestfulService, ConditionModel, ConceptModel, CommonFunctions) {
+    var self = this;
+
+    $scope.patientUuid = null;
+    $scope.condition = null;
+    const INACTIVE_STATUS = 'INACTIVE';
+
+    // this is required inorder to initialize the Restangular service
+    RestfulService.setBaseUrl('/' + OPENMRS_CONTEXT_PATH + '/ws/rest/emrapi');
+
+    /**
+     * Perform POST save
+     * PS: Make sure the 'Non Coded Uuid' setting (under Settings->Condition List) is set inorder to work with noncoded concepts,
+     * the LOCALE needs to be set (most concepts have 'en')
+     * @type {Function}
+     */
+    self.saveCondition = self.saveCondition || function () {
+        if($scope.condition.status == INACTIVE_STATUS && $scope.condition.onSetDate > $scope.condition.endDate){
+                emr.errorAlert("End Date can't be before Onset Date");
+        }
+        else{
+            var conditions = [];
+            conditions.push($scope.condition);
+            RestfulService.post('condition', conditions, function (data) {
+                emr.successAlert("Condition edited successfully")
+                window.location = '/' + OPENMRS_CONTEXT_PATH + '/coreapps/conditionlist/manageConditions.page?patientId=' + $scope.patientUuid + '&';
+            }, function (error) {
+                emr.errorAlert("Error Saving condition");
+            });
+        }
+    }
+
+    self.initCondition = self.initCondition || function () {
+            $scope.patientUuid = CommonFunctions.extractUrlArgs(window.location.search)['patientId'];
+            $scope.condition = new ConditionModel($scope.patientUuid)
+            $scope.condition = CommonFunctions.extractUrlArgs(window.location.search)['condition'];
+            self.convertURLParameterToJSON();
+
+            $scope.concept = {
+                display: $scope.condition.concept.name,
+                concept: {
+                    display: $scope.condition.concept.name,
+                },
+                conceptName: {
+                    display: $scope.condition.concept.name,
+                }
+            }
+
+            var onsetDatePicker = angular.element(document.getElementsByName('conditionStartDate'))[0];     
+            onsetDatePicker.value =  new Date($scope.condition.onSetDate).toISOString().slice(0,10);
```

Comment:
What is toISOString().slice(0,10) accomplishing?

Reply:
toISOString() is an inbuilt function which gives date in format: 2019-07-09T07:53:53.316Z . I am retrieving the first 10 characters to get date.

Reply:
Have you looked at how the rest of the reference application deals with dates?
Did you take a look at the date links that you i gave you? Not had any response from you in regards to it.

Reply:
Yes I did, but I couldn't understand how I could use the java functions in javascript so I used this inbuilt function. Even in the talk link there was no mention of how it could be used in javascript files.

Reply:
Which links did you look at?

Reply:
Also the the date being fetched from url is in milliseconds and I will have to convert it to date object before I could pass it to java file. So I will anyway have to use inbuilt function before parsing data so I thought making changes here itself would reduce the complexity.

Reply:
> Which links did you look at?

All the links you sent

Reply:
Have you looked at how the rest of the reference application is dealing with this?

Reply:
In a few js they are using some inbuilt functions and in some js files they are defining functions to handle it.

Reply:
Did you evaluate reusing those functions?

Reply:
There is a js file common.functions.js where in is some function for date is defined.I don't see any file using it, however I will try using it in this file.